stan/math/prim/err/elementwise_check.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef STAN_MATH_PRIM_ERR_ELEMENTWISE_ERROR_CHECKER_HPP
+#define STAN_MATH_PRIM_ERR_ELEMENTWISE_ERROR_CHECKER_HPP
+
+#include <stan/math/prim/err/throw_domain_error.hpp>
+#include <stan/math/prim/fun/get.hpp>
+#include <stan/math/prim/fun/size.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+#include <string>
+#include <sstream>
+#include <vector>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+template <typename T_is_good, typename T_exception>
+struct Checker {
+  const T_is_good& is_good;
+  const char* function;
+  const char* name;
+  const char* suffix;
+
+  void raise_error_impl(std::stringstream& ss) { throw T_exception{ss.str()}; }
+
+  template <typename... M>
```

Comment:
These next three functions can probably be replaced with one where `T&&` is the type of `message`.  That might not work for the `char*` version, but will definitely work for the arithmetic types and should also be generalized so that there's not ambiguity on other arithmetic types.

Reply:
They can all be replaced with one function. I broke them out specifically to avoid being general with the types. The only things that get passed to raise_error are const char*, double, and size_t.

I went ahead and replaced them.